packages/frontend/src/components/chart/configure/effects/handleEffect.ts

```diff
@@ -0,0 +1,106 @@
+import { Message } from '../messages'
+import {
+  Effect,
+  FetchActivityEffect,
+  FetchAggregateTvlEffect,
+  FetchAlternativeTvlEffect,
+  FetchTokenTvlEffect,
+} from './effects'
+
+export function handleEffect(
+  effect: Effect,
+  dispatch: (message: Message) => void,
+) {
+  switch (effect.type) {
+    case 'FetchAggregateTvl':
+      return handleFetchAggregateTvl(effect, dispatch)
+    case 'FetchAlternativeTvl':
+      return handleFetchAlternativeTvl(effect, dispatch)
+    case 'FetchTokenTvl':
+      return handleFetchTokenTvl(effect, dispatch)
+    case 'FetchActivity':
+      return handleFetchActivity(effect, dispatch)
+  }
+}
+
+function handleFetchAggregateTvl(
+  { url, requestId }: FetchAggregateTvlEffect,
+  dispatch: (message: Message) => void,
+) {
+  timeoutLoader(requestId, dispatch)
+  fetchThenDispatch(
+    url,
+    dispatch,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    (data) => ({ type: 'AggregateTvlLoaded', requestId, data }),
+    () => ({ type: 'AggregateTvlFailed', requestId }),
+  )
+}
+
+function handleFetchAlternativeTvl(
+  { url, requestId }: FetchAlternativeTvlEffect,
+  dispatch: (message: Message) => void,
+) {
+  timeoutLoader(requestId, dispatch)
+  fetchThenDispatch(
+    url,
+    dispatch,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    (data) => ({ type: 'AlternativeTvlLoaded', requestId, data }),
+    () => ({ type: 'AlternativeTvlFailed', requestId }),
+  )
+}
+
+function handleFetchTokenTvl(
+  { url, requestId, token }: FetchTokenTvlEffect,
+  dispatch: (message: Message) => void,
+) {
+  timeoutLoader(requestId, dispatch)
+  fetchThenDispatch(
+    url,
+    dispatch,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    (data) => ({ type: 'TokenTvlLoaded', requestId, token, data }),
+    () => ({ type: 'TokenTvlFailed', requestId }),
+  )
+}
+
+function handleFetchActivity(
+  { url, requestId }: FetchActivityEffect,
+  dispatch: (message: Message) => void,
+) {
+  timeoutLoader(requestId, dispatch)
+  fetchThenDispatch(
+    url,
+    dispatch,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    (data) => ({ type: 'ActivityLoaded', requestId, data }),
+    () => ({ type: 'ActivityFailed', requestId }),
+  )
+}
+
+function fetchThenDispatch(
+  url: string,
+  dispatch: (message: Message) => void,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Not necessary now, but would be nice to not have this - maybe a zod(-like) parser that makes sure types are correct?

Reply:
I agree that it would be a nice to have in the future, but this PR is already large enough. I can add a task to the backlog